ci-operator/step-registry/cucushift/installer/check/gcp/user-labels/filestore-csi/cucushift-installer-check-gcp-user-labels-filestore-csi-commands.sh

```diff
@@ -0,0 +1,279 @@
+
+  oc create -f $snapshot_manifest_filename
+  if [[ $? -ne 0 ]]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to create snapshot resources."
+    exit_code=1
+  fi
+
+  return $exit_code
+}
+
+# delete filestore csi resources
+function delete_filestore() {
+  oc delete -f $snapshot_manifest_filename
+  if [[ $? -ne 0 ]]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to delete snapshot resources."
+    exit_code=1
+  fi
+  echo "$(date -u --rfc-3339=seconds) - Sleep for 60 seconds..." && sleep 60
+
+  oc delete -f $pod_manifest_filename
+  if [[ $? -ne 0 ]]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to delete pod resources."
+    exit_code=1
+  fi
+  echo "$(date -u --rfc-3339=seconds) - Sleep for 60 seconds..." && sleep 60
+    
+  oc delete -f $pvc_manifest_filename
+  if [[ $? -ne 0 ]]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to delete pvc resources."
+    exit_code=1
+  fi    
+  echo "$(date -u --rfc-3339=seconds) - Sleep for 300 seconds..." && sleep 300
+    
+  oc delete -f $driver_manifest_filename
+  if [[ $? -ne 0 ]]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to delete driver/storageclass/volumesnapshotclass resources."
+    exit_code=1
+  fi
+
+  return $exit_code
+}
+
+# check user labels of filestore instance
+function check_user_labels() {
+  name="$(oc -n test-filestore-labels get pvc -ojson | jq -r .items[].spec.volumeName)"
+  if [ -z "${name}" ]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to find pvc volume."
+    exit_code=1
+    return $exit_code
+  fi
+
+  current_labels=""
+  readarray -t zones < <(gcloud compute regions describe "${REGION}" --format json | jq -r .zones[])
+  for zone in "${zones[@]}"; do
+    zone=$(basename $zone)
+    gcloud filestore instances describe "${name}" --zone "${zone}" &>/dev/null
+    if [[ $? -eq 0 ]]; then
+      current_labels=$(gcloud filestore instances describe "${name}" --zone "${zone}" --format json | jq -r -c .labels)
+      break
+    fi
+  done
+
+  if [[ "${expected_labels}" != "${current_labels}" ]]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Unexpected labels '${current_labels}' for '${name}'."
+    exit_code=1
+  else
+    echo "$(date -u --rfc-3339=seconds) - Matched labels '${current_labels}' for '${name}'."
+  fi
+
+  return $exit_code
+}
+
+echo "$(date -u --rfc-3339=seconds) - Creating the manifests files for creating filestore csi resources..."
+generate_manifest_files "create"
+
+echo "$(date -u --rfc-3339=seconds) - Creating pd csi resources..."
+create_filestore || echo "$(date -u --rfc-3339=seconds) - ERROR: Creating filestore csi resources."
+
+echo "$(date -u --rfc-3339=seconds) - Sleep for 600 seconds before checking filestore instance user labels..." && sleep 600
+check_user_labels || echo "$(date -u --rfc-3339=seconds) - ERROR: Checking filestore instance user labels."
```

Comment:
If `check_user_labels` failed, the following code would be still continued? If yes, the only checking way is dig into the log file, check the output?

Reply:
Updated the validation, and the step would be with non-zero exit code if validation failure. Thanks!